internal/extproc/backendauth/azure.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package backendauth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	extprocv3 "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+
+	"github.com/envoyproxy/ai-gateway/filterapi"
+)
+
+type azureHandler struct {
+	azureAccessToken string
+}
+
+func newAzureHandler(azureAuth *filterapi.AzureAuth) (Handler, error) {
+	if azureAuth != nil {
+		content, err := os.ReadFile(azureAuth.Filename)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read azure access token file: %w", err)
+		}
+		// Extract access token from secret which content a key-value string
+		// such as `azure_access_token: secret_value`.
+		lines := strings.Split(string(content), "\n")
+		var azureAccessToken string
+		for _, line := range lines {
+			parts := strings.SplitN(line, "=", 2)
+			if len(parts) == 2 && strings.TrimSpace(parts[0]) == "azure_access_token" {
+				azureAccessToken = strings.TrimSpace(parts[1])
+				break
+			}
+		}
```

Comment:
AWS credential has specific data format which AWS SDK expects, for azure it is simply a bear token which is more closer to the [APIKey](https://github.com/envoyproxy/ai-gateway/blob/main/internal/extproc/backendauth/api_key.go), so we do not need to parse the file and the secret file content can just include the token.